app/actors/hyrax/actors/file_actor.rb

```diff
@@ -5,62 +5,66 @@ class FileActor
       attr_reader :file_set, :relation, :user
 
       # @param [FileSet] file_set the parent FileSet
-      # @param [String] relation the type/use for the file.
+      # @param [Symbol, #to_sym] relation the type/use for the file
       # @param [User] user the user to record as the Agent acting upon the file
       def initialize(file_set, relation, user)
         @file_set = file_set
-        @relation = relation
+        @relation = relation.to_sym
         @user = user
       end
 
-      # Puts the uploaded content into a staging directory. Then kicks off a
-      # job to ingest the file into the repository, then characterize and
-      # create derivatives with this on disk variant.
-      # TODO: create a job to monitor this directory and prune old files that
-      # have made it to the repo
-      # @param [File, ActionDigest::HTTP::UploadedFile, Tempfile] file the file to save in the repository
-      # @param [Boolean] asynchronous set to true if you want to launch a new background job.
-      def ingest_file(file, asynchronous)
-        method = if asynchronous
-                   :perform_later
-                 else
-                   :perform_now
-                 end
-
-        IngestFileJob.send(method,
-                           file_set,
-                           working_file(file),
-                           user,
-                           ingest_options(file))
+      # Spawns a job to characterize and create derivatives.
+      # @param [Hydra::Derivatives::IoDecorator] io the file to save in the repository, with mime_type and original_name
+      # @return [Boolean] true on success (note: does NOT mean spawned job completed), false otherwise
+      # @example for File
+      #   file_actor = Hyrax::Actors::FileActor.new(fileset, :original_file, user)
+      #   file = File.open('/tmp/mydir/upload.jpg', 'rb')
+      #   io = Hydra::Derivatives::IoDecorator.new(file, 'image/jpeg', 'spirostomum_2400x1200.jpg')
+      #   file_actor.ingest_file(io)
+      # @example for Tempfile
+      #   tempfile = get_tempfile_from_request # presume you get a Tempfile from Rack, Rails or wherever
```

Comment:
Agree that `get_` is generally non-idiomatic Ruby, though the example here isn't egregious because it's not using `get` in a `getter` context. It's using `get` like `extract`. I think this is acceptable.